src/main/java/com/labsynch/labseer/service/LotServiceImpl.java

```diff
@@ -285,12 +292,23 @@ public String appendSaltCode(String corpName, Lot lot) {
 
 	@Override
 	public String generateParentLotName(Lot lot) {
```

Comment:
I was going to approach this problem differently at first and refactored the `generateParentLotName` method to fit that approach.  I changed course but left this refactor in because I thought it could be beneficial and is clearer.